### ISSUE TITLE
RPCAPI does not accept empty IP

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#702]
This issue is based on the findings in issue zonemaster/zonemaster#925.

In method `start_domain_test` a name server object optionally be specified. If specified, and non-empty, it contains a list of name servers. For each name server a name, `ns`, must be specified. One IP address, `ip` can optionally be specified. Backend correctly accepts `ns` without `ip` but it incorrectly rejects empty `ip`. Those two cases should be considered to be equal. E.g.

```
    "nameservers": [
      {
        "ip": "",
        "ns": "ns3.nic.se"
      }
    ],
```
and

```
    "nameservers": [
      {
        "ns": "ns3.nic.se"
      }
    ],
```

Example from the API document.

```
    "nameservers": [
      {
        "ip": "2001:67c:124c:2007::45",
        "ns": "ns3.nic.se"
      },
      {
        "ip": "192.93.0.4",
        "ns": "ns2.nic.fr"
      }
    ],
```

[API]: https://github.com/zonemaster/zonemaster-backend/blob/master/docs/API.md#api-method-start_domain_test

**Comment by mattias-p** [comment]
This does not add any expressiveness to the API - only [TIMTOWTDI](https://en.wikipedia.org/wiki/There%27s_more_than_one_way_to_do_it). In that light I don't think we should do it.

**Comment by matsduf** [comment]
The API does not specify what it means. The `zmb` sends either of these two when IP address is not specified:

```
"nameservers":[
  {
    "ip":"",
    "ns":"ns.nic.se"
  }
]

"nameservers":[
  {
    "ns":"ns.nic.se",
    "ip":null
  }
]
```

**Comment by mattias-p** [comment]
Agreed and agreed. I created #704 regarding the API documentation. And the `zmb` CLI interface is somewhat sketchy. In particular the `zmb start_domain_test --nameserver` parameter is really bad. I'll create an issue to get it into shape, but I need to think a little about how to draw up requirements for it.

**Comment by matsduf** [comment]
The `ip` is checked in two places in the code (RPCAPI.pm and Validator.pm). In the first place empty IPs are accepted, and that was the original place where it was checked, I think. In the second place empty IPs are not accepted.

In neither place `null` is accepted.

**Comment by matsduf** [comment]
Resolved by #705. This could be seen as a fix to resolved the issue reported in zonemaster/zonemaster#925. rather than long-term solution. Also see #703.
